subprojects/composite-builds/src/main/java/org/gradle/composite/internal/DefaultRootBuildState.java

```diff
@@ -121,12 +123,14 @@ public <T> T run(Function<? super BuildTreeLifecycleController, T> action) {
             try {
                 GradleInternal gradle = getBuildController().getGradle();
                 DefaultDeploymentRegistry deploymentRegistry = gradle.getServices().get(DefaultDeploymentRegistry.class);
-                gradle.addBuildListener(new InternalBuildAdapter() {
-                    @Override
-                    public void buildFinished(BuildResult result) {
-                        deploymentRegistry.buildFinished(result);
-                    }
-                });
+                DeprecationLogger.whileDisabled(() ->
+                    gradle.addBuildListener(new InternalBuildAdapter() {
```

Comment:
❌ `InternalBuildAdapter` shouldn't trigger a deprecation warning since it's an `InternalBuildListener`

Reply:
Please see https://github.com/gradle/gradle/pull/30000/commits/b38f0260a149fef41a79322168a2ea21bab3887d